src/BaGet.Core/Decompiler/SourceCode/ISourceCodeProvider.cs

```diff
@@ -0,0 +1,10 @@
+using BaGet.Core.Entities;
+using ICSharpCode.Decompiler.TypeSystem;
+
+namespace BaGet.Core.Decompiler.SourceCode
+{
+    internal interface ISourceCodeProvider
+    {
+        bool TryFillSources(IModule module, SourceCodeAssembly assembly);
```

Comment:
Instead of accepting a `SourceCodeAssembly`, could this accept a `SourceCodeType`?  Also, could we make it so that `ISourceCodeProvider.TryFillSources` returns `true` only if `SourceCodeType.CSharp` was filled in?

With these changes, `AssemblyDecompilerService` would play a little nicer if there are multiple source code providers

Reply:
True. I'm a bit worried about how to find sources from embedded / linked sources, since there may not be a 1:1 relationship between files and types... But that may be resolved later.